src/services/TestResultsService.ts

```diff
@@ -128,38 +139,83 @@ export class TestResultsService {
   public updateTestResult(systemNumber: string, payload: ITestResult, msUserDetails: IMsUserDetails) {
     this.removeNonEditableAttributes(payload);
     let validationSchema = this.getValidationSchema(payload.vehicleType, payload.testStatus);
-    validationSchema = validationSchema!.keys({countryOfRegistration: Joi.string().valid(countryOfRegistration).required().allow("", null)});
-    validationSchema = validationSchema!.optionalKeys(["testStationType", "testerEmailAddress", "testEndTimestamp", "systemNumber", "vin"]);
+    const testTypesValidationErrors = this.validateTestTypes(payload);
+    if (testTypesValidationErrors.length) {
+      return Promise.reject(new HTTPError(400, {errors: testTypesValidationErrors}));
+    }
+    // temporarily remove testTypes to validate only vehicle details and append testTypes to the payload again after the validation
+    const {testTypes} = payload;
+    delete payload.testTypes;
```

Comment:
I don't understand why we are temporarily removing that

Reply:
please see the reply to Umer's comment above